types/src/error.rs

```diff
@@ -38,6 +38,14 @@ pub enum CallError {
 	},
 }
 
+// NOTE(niklasad1): this `From` impl is a bit opinionated to regard all generic errors as `CallError`.
+// The most common use case is when register method calls on the servers.
+impl From<anyhow::Error> for Error {
```

Comment:
the annoying thing with this is that if a user provides their own error type:

then one have to this:

```rust
let err = MyError;
// This might not work because rustc can't deduce the first `into` as anyhow::Error
let call_err: CallError = err.into().into();

// Then users of this API must use `anyhow::anyhow!` themselves such as (that we forces them to do):
let call_err: CallError = anyhow!(err).into();
```

Reply:
at least this was the case when I tested it in substrate.

Reply:
alternatively, we could have helper function:

```rust
impl CallError {
   fn from_std_error<std::error::Error + Send + Sync + 'static>(err: E) -> CallError {
     CallError::Failed(e)
   }
}
```

Reply:
Does it need to be `anyhow::Error`? Can't we just accept anything that's `std::fmt::Debug`?

Reply:
I guess so but we would need to change it to a `String` [here](https://github.com/paritytech/jsonrpsee/blob/master/types/src/error.rs#L28) then or what's your idea?